app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,76 @@
+class PostsController < ApplicationController
+  before_action :set_post, only: %i[show edit update destroy]
+  before_action :authenticate_user!, except: %i[index show]
+
+  # GET /posts
+  # GET /posts.json
+  def index
+    @posts = Post.all
+  end
+
+  # GET /posts/1
+  # GET /posts/1.json
+  def show; end
+
+  # GET /posts/new
+  def new
+    @post = current_user.posts.build
+  end
```

Comment:
Initialize an empty post this way `@post = Post.new` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_new_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication)) This is the preferred way. It also avoids repetition of the `build` method in both the `new` and `create` actions.

Reply:
I'm using a before_action to restrict access to some methods including the :new and :create methods, devise give a helper and I'm using this helper.
![image](https://user-images.githubusercontent.com/39846015/101105908-c28b2000-359c-11eb-96d3-eb6417e30992.png)